diff -ur eigenpy-1.6.5/cmake/boost.cmake eigenpy-1.6.5.patched/cmake/boost.cmake
--- eigenpy-1.6.5/cmake/boost.cmake	2019-11-09 11:54:43.000000000 +0100
+++ eigenpy-1.6.5.patched/cmake/boost.cmake	2019-11-15 23:36:35.675750569 +0100
@@ -100,7 +100,9 @@
 
       SET(BOOST_PYTHON_COMPONENT_LIST)
       # Test: python2 or python3
-      LIST(APPEND BOOST_PYTHON_COMPONENT_LIST "python${PYTHON_VERSION_MAJOR}") 
+      LIST(APPEND BOOST_PYTHON_COMPONENT_LIST "python${PYTHON_VERSION_MAJOR}")
+      # Test: python2_x or python3_x needed for Gentoo comaptibility
+      LIST(APPEND BOOST_PYTHON_COMPONENT_LIST "python${PYTHON_VERSION_MAJOR}_${PYTHON_VERSION_MINOR}") 
       # Test: python2x or python3x
       LIST(APPEND BOOST_PYTHON_COMPONENT_LIST "python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}") 
       # Test: python-py2x or python-py3x
@@ -256,7 +258,8 @@
       GET_FILENAME_COMPONENT(LIB_NAME ${LIB_PATH} NAME)
       PKG_CONFIG_APPEND_LIBS_RAW("-l${LIB_NAME}")
     ELSE()
-      GET_FILENAME_COMPONENT(LIB_NAME ${LIB_PATH} NAME_WE)
+      GET_FILENAME_COMPONENT(LIB_NAME ${LIB_PATH} NAME)
+      STRING(REGEX REPLACE ".so" "" LIB_NAME "${LIB_NAME}")
       STRING(REGEX REPLACE "^lib" "" LIB_NAME "${LIB_NAME}")
       PKG_CONFIG_APPEND_LIBS_RAW("-l${LIB_NAME}")
     ENDIF(APPLE)
